### ISSUE TITLE
Action WF refinements

### ISSUE THREAD

**Issue by riccardoporreca** [miraisolutions/techguides#13]
- switch to `ruby/setup-ruby`
  - this is now the starter workflow for Ruby, making actions/setup-ruby deprecated (see actions/setup-ruby#80)..
  - Consider bundler setup to leverage the (proper) caching are implicitly provided

As an alternative, use a containerized HTMLProofer (Docker has out-of-the-box support in Actions)
- https://github.com/gjtorikian/html-proofer#using-through-docker

**Comment by Chargothrond** [comment]
Tried using the docker image directly, which seems to work fine. However, there is a new htmlproofer warning about images missing alt texts.

I noticed that the global fig_caption option doesn't work with git-books, but the fig.alt chunk option would by now. I prefer however the trick with appending "\". See these links for context:
- "\" workaround: https://community.rstudio.com/t/have-an-alternative-text-but-no-caption-or-a-different-caption-in-bookdown-git-book/50370
- fig.alt chunk option apparently works now (didn't test it): https://github.com/rstudio/bookdown/issues/856
- https://bookdown.org/yihui/rmarkdown-cookbook/html-accessibility.html provides some useful information as well

Alternatively, it would also be possible to ignore the alt warnings, but I decided that these warnings actually point out a valid issue and that we should put these alt texts. These links that I found might have helped with this approach:
- https://www.petefreitag.com/item/903.cfm
- https://stackoverflow.com/questions/42777355/cannot-get-htmlproofer-to-ignore-missing-image-alt-tags

**Comment by riccardoporreca** [comment]
@Chargothrond, will look into the details... Note that in the existing action (not Docker-based) we are have been successfully ignoring img alt thing (which is something inherited from the inital setup with Travis, see https://github.com/miraisolutions/techguides/commit/3505fedd4e0a92808fcccc37a8e765b0bdd1880e). I would keep the ignore, which will also give us the chance to test passing command-line arguments to the docker image.

Can you open a dedicated issues to fix the alt problem, pasting there the comment above. This way we can focus on the workflow refinement as part of this one

**Comment by Chargothrond** [comment]
@riccardoporreca, don't see the necessity to create a new issue for this, can we not just document it here? We can also rename the title. I find it more difficult to find relevant information if it is distributed across many issues, branches and PRs.

Either way, I have tested ignoring them on the latest commit: https://github.com/miraisolutions/techguides/runs/2227362485
While the option is passed to the docker command in blue, it isn't applied for some reason. I am too busy with ZIG stuff to investigate this still this week.

Personally, I'd prefer to just wrap this up by actually fixing those alt issues (i.e. create a PR on my previous commit). Given the logs, it also looks like passing the option to the docker command works fine. But feel free to use https://github.com/miraisolutions/techguides/tree/feature/13-action-wf-refinements if you want to investigate ignoring the alt attribute further.

For reference, I copy here what I used in the site.yaml to pass an option to that docker container:
      - name: Test website
        uses: docker://klakegg/html-proofer:3.18.8
        with:
          args: --alt-ignore '/.*/'
